### DISCUSSION TITLE
What are the benefits of using react-rails over hosting a SPA with Puma?

### DISCUSSION THREAD

**Discussion by raphox** [reactjs/react-rails#1286] [Q&A]
I am considering using react-rails to integrate React with Rails, but I am not sure what are the advantages of this approach over hosting a single-page application (SPA) with Puma, the default Rails server. Can anyone explain the pros and cons of each option, and what are the best practices for using them?

Recently I wrote a post commenting about this topic here https://medium.com/@raphox/how-i-dumped-ruby-on-rails-and-eloped-with-next-js-but-its-not-what-you-re-thinking-6f9fe448cba1

**Comment by Judahmeek** [comment]
The advantage of a custom tech stack including ReactRails over Next.js is lower complexity, less lock-in, ability to use a single monolith repository, & greater ability to experiment with other libraries.
